packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    CalldataInfo,
+    ExchangeMarketBuySmartContractParams,
+    SmartContractParamsInfo,
+    SwapQuote,
+    SwapQuoteConsumer,
+    SwapQuoteConsumerError,
+    SwapQuoteConsumerOpts,
+    SwapQuoteExecutionOpts,
+    SwapQuoteGetOutputOpts,
+} from '../types';
+import { assert } from '../utils/assert';
+import { swapQuoteConsumerUtils } from '../utils/swap_quote_consumer_utils';
+import { utils } from '../utils/utils';
+
+export class ExchangeSwapQuoteConsumer implements SwapQuoteConsumer<ExchangeMarketBuySmartContractParams> {
+
+    public readonly provider: ZeroExProvider;
+    public readonly networkId: number;
+
+    private readonly _contractWrappers: ContractWrappers;
+
+    constructor(supportedProvider: SupportedProvider, options: Partial<SwapQuoteConsumerOpts> = {}) {
+        const { networkId } = _.merge({}, constants.DEFAULT_SWAP_QUOTER_OPTS, options);
+        assert.isNumber('networkId', networkId);
+
+        const provider = providerUtils.standardizeOrThrow(supportedProvider);
+        this.provider = provider;
+        this.networkId = networkId;
+        this._contractWrappers = new ContractWrappers(this.provider, {
+            networkId,
+        });
+    }
+
+    public async getCalldataOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): Promise<CalldataInfo> {
+        assert.isValidSwapQuote('quote', quote);
+
+        const { params, to, ethAmount, methodAbi } = await this.getSmartContractParamsOrThrowAsync(quote, opts);
+        const abiEncoder = new AbiEncoder.Method(methodAbi);
+        const args = [
+            params.orders,
+            params.makerAssetFillAmount,
+            params.signatures,
+        ];
+        const calldataHexString = abiEncoder.encode(args);
+        return {
+            calldataHexString,
+            methodAbi,
+            to,
+            ethAmount,
+        };
+    }
+
+    public async getSmartContractParamsOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): Promise<SmartContractParamsInfo<ExchangeMarketBuySmartContractParams>> {
+        assert.isValidSwapQuote('quote', quote);
+
+        const { orders, makerAssetFillAmount } = quote;
+
+        const signatures = _.map(orders, o => o.signature);
+
+        const params: ExchangeMarketBuySmartContractParams = {
+           orders,
+           signatures,
+           makerAssetFillAmount,
+        };
+
+        const methodAbi = utils.getMethodAbiFromContractAbi(
+            this._contractWrappers.exchange.abi,
+            'marketBuyOrdersNoThrow',
+        ) as MethodAbi;
+
+        return {
+            params,
+            to: this._contractWrappers.exchange.address,
+            methodAbi,
+        };
+    }
+
+    public async executeSwapQuoteOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteExecutionOpts>): Promise<string> {
+        assert.isValidSwapQuote('quote', quote);
+
+        const { takerAddress, gasLimit, gasPrice } = opts;
```

Comment:
No default values here?

Reply:
These values are all optional so there is no need for a default value obj.